pkg/interface/chat/src/js/components/lib/chat-input.js

```diff
@@ -151,14 +150,35 @@ export class ChatInput extends Component {
       return;
     }
 
-    let letter = this.getLetterType(state.message);
+    let URLs = [];
+    let letter = state.message.split(" ").map((each) => {
+      if (this.isUrl(each)) {
+        let URL = this.getLetterType(each);
+        URLs.push(URL);
+      }
+      else {
+        return each;
+      }
+    }).join(" ");
+
+    letter = this.getLetterType(letter);
 
     props.api.chat.message(
       props.station,
       `~${window.ship}`,
       Date.now(),
       letter
     );
+
+    for (let each of URLs) {
+      props.api.chat.message(
+        props.station,
+        `~${window.ship}`,
+        Date.now(),
+        each
+      );
+    }
```

Comment:
I thought the intended behavior was, for a message like "hey i like https://urbit.org it's really cool", to send as three messages in the following order:

- `%text` hey i like
- `%url` https://urbit.org
- `%text` it's really cool

Looks like the current implementation always sends with `%url`s after smooshing the rest of the content into a single `%text`, regardless of where they appear in the text. In the above example, that gets you "hey i like it's really cool". Seems not good?